### ISSUE TITLE
Accessories are not accessible from internet through Home Hub

### ISSUE THREAD

**Issue by Adam-Langley** [ikalchev/HAP-python#265]
I have an Apple TV set up to act as my Home Hub.
I am running the HAP-Python examples on a Raspberry Pi, connected to wifi.
This works for retail HomeKit devices, as well as custom HomeKit software implementations using the software library https://github.com/brutella/hc

The example HAP-python accessories work locally, but not through the Home Hub. The HAP-python server never receives any incoming requests when access is attempted from the internet.

Has anyone proven otherwise?

**Comment by rblenkinsopp** [comment]
This seems to work correctly for me (I'm using a iPad Pro as my home hub). I haven't done anything special that I'm aware of to make this work.

**Comment by Adam-Langley** [comment]
Thanks @rblenkinsopp - I find that it works if running on a Mabook Pro (wifi), but not rpi (wifi). I am guessing something in the networking layer of the O/S is to blame - I didn't want to got down that path unless someone could confirm it worked on their Raspberry Pi.
Can I ask, whats your Pi hardware/OS/etc?

**Comment by rblenkinsopp** [comment]
This was working correctly for me on a Raspberry Pi Zero W, running Raspberry Pi OS Lite (Rasbian Lite) I've not done anything to it other than setting up WPA Supplicant, SSH and installing the Python libraries.

**Comment by Adam-Langley** [comment]
Hi @rblenkinsopp - I found what my specific issues were.

Firstly, I am using a wifi repeater which appears to make zerconf broadcasts very unreliable. This affects the registration stage more than anything else - but it is a problem regardless.
Secondly, I had removed some resolutions from my camera configuration. Specifically, when on WIFI, the HomeKit app will request higher available resolutions, and when accessing via a Home Hub, it will attempt lower resolutions. If the lower resolutions (640x360, 640x480, 480x360, etc) were not available _at initial registration time_ then the video stream will not be initialized.

So, the trick is - if you change the resolutions from what is in the examples, make sure you have at least 1 of the following resolutions if you want video to work through a Home Hub (this is also the order of precedence, for me anyway):

1. 640x360
1. 640x480
1. 480x360

Note: changing the available resolutions _after you have registered the camera accessory with homekit_ will have no effect. Homekit will not see these changes.
